pkg/profile/artifact/artifact.go

```diff
@@ -5,13 +5,33 @@ import (
 	"sigs.k8s.io/kustomize/api/types"
 )
 
+// Object is an artifact object which adds path to a runtime object.
+// path adds location detail to help generate this object.
+type Object struct {
+	runtime.Object
+	// Path, if set, will be used as folder location for this Object.
+	Path string
+	// Name, if given, will overwrite what this object's filename representation should be.
+	Name string
+}
+
+// Kustomize contains resources which direct flux to what to install.
+type Kustomize struct {
+	// LocalResourceLimiter resource which limits the number of objects scanned by flux.
+	LocalResourceLimiter *types.Kustomization
+	// ObjectWrapper is the top most Kubernetes object wrapper for all resources.
+	ObjectWrapper *types.Kustomization
+}
+
```

Comment:
I still find this really confusing :sweat_smile: but I'm honestly struggling to think of something better. Short of calling them `KubernetesKustomize` and `FluxKustomize` :joy:

Reply:
The problem with that is that both are there because of Flux. :D